### PR TITLE
Various style/compatibility fixes

### PR DESCRIPTION
**Pull request by tibetiroka**
This PR fixes a number of issues on the website, namely:
- missing `lang` attribute for the html tag
- missing `alt` for images
- extra `/` on void tags (img, link etc.)
- JavaScript issues (using `let`, `==` etc.)
- CSS unit specifiers

Hopefully, Warp Magic™ will be available for testing.

### CODE REVIEW

**Comment by warp-core** [comment]
It builds and stuff.

File: _posts/2016-09-08-designing-ship-sprites.md [hunk 1]
```diff
@@ -37,15 +37,15 @@ The final step is to export an image and post-process it. To keep things simple,
 
 Before doing any post-processing, test out different sprite sizes in the game to see what looks good and fits with the existing ships. Then, re-render the image at exactly 2 or 4 times the intended sprite size. Also make sure the @1x image will have at least a 1-pixel blank border on all sides so that the outline will render correctly.
 
-<img class="centered shadowed" src="/images/blog/summer-leaf-quadrants.jpg" width="400" height="700" />
+<img class="centered shadowed" src="/images/blog/summer-leaf-quadrants.jpg" width="400" height="700" alt="A Summer Leaf with divided into four quadrants, with different effects shown on each">
```
Comment (mOctave) [inline _posts/2016-09-08-designing-ship-sprites.md:40]:
```suggestion
<img class="centered shadowed" src="/images/blog/summer-leaf-quadrants.jpg" width="400" height="700" alt="A Summer Leaf divided into four quadrants, with different effects shown on each">
```

File: _posts/2017-01-17-unstable-release-0.9.5.md [hunk 2]
```diff
@@ -11,6 +11,6 @@ The biggest change to the game mechanics in this release is the addition of a de
 
 The way depreciation is implemented is that every ship and outfit loses value each day until it is 360 days old, at which point it is worth 25% of its original value. If you sell it, it stays "in stock" and can be bought back at the depreciated price until you leave the planet. Captured ships and plundered outfits always start out fully depreciated. Depreciation does not in any way affect the performance of a ship or outfit; a brand new ship is no more powerful or fast than a fully depreciated one.
 
-<img class="centered shadowed" src="/images/depreciation.png" width="500" height="400" />
+<img class="centered shadowed" src="/images/depreciation.png" width="500" height="400" alt="A graph showing how the value of ships decreases as they are kept for longer, with 50% of the value being around 80 days.">
```
Comment (mOctave) [inline _posts/2017-01-17-unstable-release-0.9.5.md:14]:
```suggestion
<img class="centered shadowed" src="/images/depreciation.png" width="500" height="400" alt="A graph showing how the value of ships decreases as they are kept for longer, having about 50% of their original value after 80 days">
```

They should either all have periods, or none of them.

File: _posts/2017-02-06-alien-sociology.md [hunk 3]
```diff
@@ -7,10 +7,10 @@ Sociologists studying the Quarg and other alien civilizations have noted that th
 
 Advanced species, on the other hand, are often wildly different from each other and behave in ways that seem inscrutable or even irrational to outside observers. The currently accepted explanation for this is that all sentient species initially face the same threats to their survival: lack of food, extreme environmental conditions, attacks by other tribes or by other creatures, etc. In order to survive this stage each species will generally settle into one of a limited number of near-optimal strategies for facing those threats and satisfying what xeno-sociologists refer to as the "common hierarchy" of needs:
 
-<img class="centered" src="/images/alien-sociology.png" width="420" height="380" />
+<img class="centered" src="/images/alien-sociology.png" width="420" height="380" alt="A graph showing a breakdown of alien sociology. On the left, it shows crisis, stagnation, reorientation, the idiosyncratic hierarchy of vocation on top, and the common hierarchy of needs on the bottom.">
```
Comment (mOctave) [inline _posts/2017-02-06-alien-sociology.md:10]: Nonsensical

File: _posts/2017-09-23-remastered-ship-graphics.md [hunk 4]
```diff
@@ -5,12 +5,12 @@ author: Michael Zahniser
...
 
-<img class="centered shadowed" src="/images/blog/mereti.jpg" width="540" height="540" />
+<img class="centered shadowed" src="/images/blog/mereti.jpg" width="540" height="540" alt="A collection of Kor Mereti ships looking blocky">
 
-<img class="centered shadowed" src="/images/blog/wanderers.jpg" width="540" height="440" />
+<img class="centered shadowed" src="/images/blog/wanderers.jpg" width="540" height="440" alt="Various Wanderer ships showing parts with different thickness">
 
-<img class="centered shadowed" src="/images/blog/sestor.jpg" width="540" height="540" />
+<img class="centered shadowed" src="/images/blog/sestor.jpg" width="540" height="540" alt="Kor Sestor ships with 3D-like exterior plating"/>
```
Comment (mOctave) [inline _posts/2017-09-23-remastered-ship-graphics.md:12]: You've missed some `/`s.

File: ship_builder.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 ---
-<html>
+<html lang="en">
```
Comment (mOctave) [inline ship_builder.html:3]: `lang="en-US"` might be better, although it's okay as is.

File: sounds.html [hunk 6]
```diff
@@ -1,10 +1,10 @@
-<html><head><title>Vote on sound effects you like</title>
+<html lang="en"><head><title>Vote on sound effects you like</title>
```
Comment (mOctave) [inline sounds.html:1]: Same here.

**Review by mOctave — CHANGES_REQUESTED** [review]
Just a couple of errors...

ps Also, although common practice is to remove all trailing slashes on void tags, I personally find that they help distinguish them from other tags, and in this context they shouldn't break anything. Just food for thought.

File: ship_builder.html (hunk 5 above)
Comment (tibetiroka) [reply ship_builder.html:3]:
```suggestion
<html lang="en-US">
```

File: sounds.html (hunk 6 above)
Comment (tibetiroka) [reply sounds.html:1]:
```suggestion
<html lang="en-US"><head><title>Vote on sound effects you like</title>
```

File: _posts/2017-02-06-alien-sociology.md (hunk 3 above)
Comment (tibetiroka) [reply _posts/2017-02-06-alien-sociology.md:10]:
```suggestion
<img class="centered" src="/images/alien-sociology.png" width="420" height="380">
```
Yeah I'm just going to remove this

File: _posts/2017-09-23-remastered-ship-graphics.md (hunk 4 above)
Comment (tibetiroka) [reply _posts/2017-09-23-remastered-ship-graphics.md:12]:
```suggestion
<img class="centered shadowed" src="/images/blog/sestor.jpg" width="540" height="540" alt="Kor Sestor ships with 3D-like exterior plating">
```

File: _posts/2017-09-23-remastered-ship-graphics.md [hunk 7]
```diff
@@ -5,12 +5,12 @@ author: Michael Zahniser
...
 
-<img class="centered shadowed" src="/images/blog/navy.jpg" width="540" height="440" />
+<img class="centered shadowed" src="/images/blog/navy.jpg" width="540" height="440" alt="Navy ships with a variety of colored panels"/>
```
Comment (tibetiroka) [inline _posts/2017-09-23-remastered-ship-graphics.md:14]:
```suggestion
<img class="centered shadowed" src="/images/blog/navy.jpg" width="540" height="440" alt="Navy ships with a variety of colored panels">
```

**Comment by tibetiroka** [comment]
> Also, although common practice is to remove all trailing slashes on void tags, I personally find that they help distinguish them from other tags, and in this context they shouldn't break anything. Just food for thought.

It's not just common practice. They have [no effect on the tag itself](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-in-void-element-start-tags-do-not-mark-the-start-tags-as-self-closing) and can [mess up parsers sometimes](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-directly-preceded-by-unquoted-attribute-values), at least according to the W3C - and they are the ones managing the HTML standard, so I'm inclined to listen to their recommendations.

**Comment by mOctave** [comment]
> They have [no effect on the tag itself](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-in-void-element-start-tags-do-not-mark-the-start-tags-as-self-closing) and can [mess up parsers sometimes](https://github.com/validator/validator/wiki/Markup-%C2%BB-Void-elements#trailing-slashes-directly-preceded-by-unquoted-attribute-values), at least according to the W3C - and they are the ones managing the HTML standard, so I'm inclined to listen to their recommendations.

It looks like we've done the same research and come to different conclusions. Before I go too far, I'd like to say that I'm  honestly okay with removing trailing slashes, as long as it's the standard way we do things and we have a good reason to do them. I'd certainly prefer no trailing slashes to the current mess.

<details><summary>With that said, here's my counterargument if you want to read it...</summary>
<p>
The first link you present literally lists three reasons <i>to</i> use trailing slashes. Sure, they don't do anything, but as I've said they make it much more obvious whether you should be looking for a closing tag or not when you're actually writing code, on top of the looking pretty/formatting/JSX reasons they provide. The page also provides a couple of reasons why trailing slashes must be misleading, but that's hardly a reason to outright avoid them as long as you're using them intentionally and understanding their insignificance.

As to the second case, may I quote `the HTML standard states that trailing slashes should be used only with caution`. They specifically only break parsers when quotes aren't used. Our pages use quotes absolutely everywhere, so that isn't a problem. W3C isn't saying "don't use trailing slashes because they break stuff," they're saying "make sure that you never use trailing slashes right after an attribute because that'll change the attribute." We're not doing that, so we should be fine.
</p>
</details>

File: _posts/2017-04-04-timeline-of-ship-graphics.md [hunk 8]
```diff
@@ -18,6 +18,6 @@ It always encourages me to read early novels by my favorite authors and discover
 
 So if your initial attempts aren't turning out the way you envisioned, don't lose hope! And, as a bonus, I'll leave you with an example of what the first outfit graphics looked like:
 
-<img class="centered shadowed" src="/images/timeline/steering.png" width="200" height="140" />
+<img class="centered shadowed" src="/images/timeline/steering.png" width="200" height="140" alt="A sprite for a steering engine with various sideways-facing exhausts">
```
Comment (mOctave) [inline _posts/2017-04-04-timeline-of-ship-graphics.md:21]: Maybe emphasize how little like a steering engine it looks? Probably fine as-is, though.

File: index.html [hunk 9]
```diff
@@ -7,33 +7,33 @@ <h2 class="header">A galaxy lies open for you to explore.</h2>
 <p>Endless Sky is a 2D space trading and combat game similar to the classic Escape Velocity series. Explore other star systems. Earn money by trading, carrying passengers, or completing missions. Use your earnings to buy a better ship or to upgrade the weapons and engines on your current one. Blow up pirates. Take sides in a civil war. Or leave human space behind and hope to find friendly aliens whose culture is more civilized than your own.</p>
 <p><a href="https://github.com/endless-sky/endless-sky/releases/latest">Downloads</a> are available for Windows, Mac OS, and Linux. Endless Sky is free and open source, the product of a growing community of developers and content creators.</p>
 
-<img class="intro-image" src="/images/introduction/starting-ships.jpg" title="The three ships you can choose between at the start of the game." />
+<img class="intro-image" src="/images/introduction/starting-ships.jpg" title="The three ships you can choose between at the start of the game.">
```
Comment (mOctave) [inline index.html:10]: Should these have alt attributes too?

**Review by mOctave — COMMENTED** [review]
Looks good now, just a couple comments.

File: index.html (hunk 9 above)
Comment (tibetiroka) [reply index.html:10]: I'm not sure. They already have titles, which isn't 100% a replacement for alts, but I'm not sure how much extra information I could fit into them.

**Comment by tibetiroka** [comment]
Yeah, you have a point there... Personally, I prefer not to use slashes on those tags, but whatever works here, I guess.

**Comment by Amazinite** [comment]
A merge conflict on the website repo. Who would've thought.

**Comment by tibetiroka** [comment]
Okay this was weird. Github did tell me there was a merge conflict, but it didn't let me fix it there, so I puled the remote changes with git and... there were no conflicts? And now there aren't any conflicts here either?

**Comment by Amazinite** [comment]
#37 changed the name of one of the files you changed the contents of. You can't fix renamed file conflicts from the browser. Merging from the master via git fixes such conflicts automatically.

**Pull request merged**